parsec/interfaces/dtd/insert_function.c

```diff
@@ -227,6 +227,7 @@ parsec_dtd_dequeue_taskpool(parsec_taskpool_t *tp)
         should_dequeue = 1;
         parsec_list_nolock_remove(tp->context->taskpool_list,
                                   (parsec_list_item_t *)tp);
+        parsec_list_item_singleton((parsec_list_item_t*)tp);
```

Comment:
Could use the macro LIST_ITEM_SINGLETON here

Reply:
At this point I'm not sure the macro remains necessary in the code base, it always defaults to calling the `parsec_list_item_singleton ` function.

Reply:
the macro does the list_item cast, that's all it does. If we think this is not useful, we should remove the macro then.

Reply:
ok, let me keep the code consistent with the rest (I'll switch to PARSEC_LIST_ITEM_SINGLETON)

Reply:
Already fixed in the current code